### PR TITLE
[logstash] feat: add podAffinity settings

### PR DESCRIPTION
**Pull request by ebuildy**
Fix https://github.com/elastic/helm-charts/issues/1032

- [X] Chart version *not* bumped (the versions are all bumped and released at the same time)
- [X] README.md updated with any new values or changes
- [X] Updated template tests in `${CHART}/tests/*.py` 
- [ ] Updated integration tests in `${CHART}/examples/*/test/goss.yaml`

### CODE REVIEW

File: logstash/README.md [hunk 1]
```diff
@@ -128,6 +128,7 @@ useful for the [http input plugin][], for instance.
...
 | `nameOverride`            | Overrides the chart name for resources. If not set the name will default to `.Chart.Name`                                                                                                                                            | `""`                                  |
 | `nodeAffinity`            | Value for the [node affinity settings][]                                                                                                                                                                                             | `{}`                                  |
+| `podAffinity`             | Value for the [pod affinity settings][]                                                                                                                                                                                              | `{}`                                  |
```
Comment (jmlrt) [inline logstash/README.md:131]: This link isn't vaild, can you add a link like in https://github.com/ebuildy/helm-charts/blob/master/logstash/README.md#L219?

File: logstash/tests/logstash_test.py [hunk 2]
```diff
@@ -556,6 +556,26 @@ def test_adding_a_node_affinity():
         ]
     }
 
+def test_adding_an_podaffinity_rule():
```
Comment (jmlrt) [inline logstash/tests/logstash_test.py:559]: NIT
```suggestion
def test_adding_a_pod_affinity_rule():
```

File: logstash/values.yaml [hunk 3]
```diff
@@ -154,9 +154,13 @@ antiAffinityTopologyKey: "kubernetes.io/hostname"
 antiAffinity: "hard"
 
 # This is the node affinity settings as defined in
-# https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#node-affinity-beta-feature
+# https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#node-affinity
```
Comment (jmlrt) [inline logstash/values.yaml:157]: Thanks for fixing this link 👍

**Review by jmlrt — CHANGES_REQUESTED** [review]
Some NIT things to fix but mostly LGTM 👍

**Comment by jmlrt** [comment]
jenkins test this please

File: logstash/README.md (hunk 1 above)
Comment (ebuildy) [reply logstash/README.md:131]: done

**Review by jmlrt — APPROVED** [review]
LGTM⛴

**Comment by jmlrt** [comment]
jenkins test this please

**Review by jmlrt — CHANGES_REQUESTED** [review]
Some changes are required to make [Black](https://black.readthedocs.io/en/stable/) formatter happy in [elastic+helm-charts+pull-request+lint-python/835](https://devops-ci.elastic.co/job/elastic+helm-charts+pull-request+lint-python/835/label=docker&&virtual&&ubuntu-18.04/console).

Can you run `black logstash/tests/logstash_test.py`?

**Comment by jmlrt** [comment]
> Some changes are required to make [Black](https://black.readthedocs.io/en/stable/) formatter happy in [elastic+helm-charts+pull-request+lint-python/835](https://devops-ci.elastic.co/job/elastic+helm-charts+pull-request+lint-python/835/label=docker&&virtual&&ubuntu-18.04/console).
> 
> Can you run `black logstash/tests/logstash_test.py`?

Black CI test can be so annoying some time, but it helps keeping consistency in test code format.

**Comment by jmlrt** [comment]
👋🏻 @ebuildy Do you think your could rebase your PR and fix the python formatting issues?

**Comment by ebuildy** [comment]
ok, I am removing this one and creating a new PR

**Comment by ebuildy** [comment]
hey @jmlrt , I have just create https://github.com/elastic/helm-charts/pull/1257 from a fresh branch. Thanks you.

**Pull request closed without merging**